### PR TITLE
[Macros] Attached macro expansions return single string 

### PR DESCRIPTION
**Pull request by rintaro**
swift-syntax change https://github.com/apple/swift/pull/66918

* Move `collapse(expansions:for:attachedTo:)` to `SwiftSyntaxMacroExpansion`
* `SwiftSyntaxMacroExpansion.expandAttachedMacro()` now perform collapsing
* `SwiftSyntaxMacroExpansion.expandAttachedMacroWithoutCollapsing()` to keep old behavior
* IPC request `getCapability` now sends the host protocol version
* Unified IPC response `macroExpansionResult` that returns single string for both `expandFreestandingMacro` and `expandAttachedMacro`
* Compiler accepts old `expandFreestandingMacroResult` and `expandAttachedMacroResult` to keep compatibility
* Plugins check the compiler's protocol version to see if it suppports `macroExpansionResult`, and fall back to old behavior if necessary

Also:
* Move `checkMacroDefinition()` to `SwiftSyntaxMacroExpansion` module

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/66918
@swift-ci Please test

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/66918
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

File: Sources/SwiftCompilerPluginMessageHandling/CompilerPluginMessageHandler.swift [hunk 1]
```diff
@@ -40,6 +40,23 @@ public protocol MessageConnection {
   func waitForNextMessage<RX: Decodable>(_ type: RX.Type) throws -> RX?
 }
 
+/// Represent the capability of the plugin host (i.e. compiler).
+struct HostCapability {
+  var protocolVersion: Int
+
+  // Create an "oldest" capability.
+  init() {
+    self.protocolVersion = 0
```
[the remark this answers is not in the document]
Comment (rintaro) [reply Sources/SwiftCompilerPluginMessageHandling/CompilerPluginMessageHandler.swift:49]: `4` doesn't mean anything in practice. If we use 4 as default, we should add doc-comment explaining that. I don't think it's worth it.

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/66918
@swift-ci Please test

**Review by DougGregor — APPROVED** [review]

**Pull request merged**